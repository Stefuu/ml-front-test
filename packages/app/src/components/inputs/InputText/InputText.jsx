import React from 'react'
import { rem } from 'polished'
import { theme } from 'styled-tools'
import tag from 'clean-tag'
import styled, { css } from 'styled-components'
import { width, maxWidth } from 'styled-system'

export default styled(({fluid, ...props}) => <tag.input {...props} />)`
    font-size: ${rem(16)};
    padding: ${rem(5)} ${rem(0)} ${rem(5)} ${rem(13)};
    ::placeholder { 
        color: ${theme('colors.b30')};
    }
    border: none;
    border-top-left-radius: ${rem(4)};
    border-bottom-left-radius: ${rem(4)};
    outline: none;
    ${props => props.fluid && css`
        width: 100%;
    `}
    ${maxWidth}
    ${width}
`
