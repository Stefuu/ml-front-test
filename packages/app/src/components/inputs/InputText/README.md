# InputText

## Default input for text
