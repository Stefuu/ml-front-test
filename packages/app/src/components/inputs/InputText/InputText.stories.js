import React from 'react'
import { storiesOf } from '@storybook/react'
import { doc } from 'storybook-readme'
import readmeFile from './README.md'
import centered from '@storybook/addon-centered'
import theme from '../../themes/default'
import { ThemeProvider } from 'styled-components'
import InputText from './InputText'
import GlobalStyle from '../../modules/GlobalStyle'
import Content from '../../modules/content'
import Flex from '../../elements/Flex'

storiesOf('app/components/inputs/InputText', module)
  .addParameters({ jest: ['InputText.test.js'] })
  .add('README', doc(readmeFile))
  .addDecorator(centered)
  .addParameters({ jest: ['Content.test.js'] })
  .add('simple', () => (
    <ThemeProvider theme={theme}>
      <Flex
        alignItems='center'
        justifyContent='center'
        as={Content}
        width={'100vw'}
        color='#FFE600'>
        <GlobalStyle />
        <InputText />
      </Flex>
    </ThemeProvider>
  ))
