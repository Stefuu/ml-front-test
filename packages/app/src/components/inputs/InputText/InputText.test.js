import React from 'react'
import { render } from 'react-testing-library'
import InputText from './InputText'

describe('InputText', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<InputText />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
