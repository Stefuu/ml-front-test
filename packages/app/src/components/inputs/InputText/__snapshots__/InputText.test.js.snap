// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText Should render [snapshot] 1`] = `
<input
  class="sc-bdVaJa kalBeQ"
/>
`;
